### DISCUSSION TITLE
[ZK Stack] - <failed L1 transaction and cannot restart >

### DISCUSSION THREAD

**Discussion by qiwenzhou** [zkSync-Community-Hub/zksync-developers#539] [ZK Stack]
### Module Affected

- [X] System contracts (era-system-contracts)
- [X] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [X] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I ran a hyperchain in my server.Something wrong went with L1 and my chain server went down. 
There is no heavy work in server.
<img width="1113" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/11c0f738-4dc7-4bdd-ac09-9666c39900fd">
<img width="773" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/ba4eaff4-217f-4e6c-b265-d301464bda31">

### Expected Behavior

run for a long time 

### Current Behavior

crushed and cannot restart.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by qiwenzhou** [comment]
Look like some disorder happened but dont know how.

**Comment by qiwenzhou** [comment]
more info:this error happened when server tried to commit batch4,but batch2,3 is uncommitted because of the transaction nonce is too low. Is is possible that this could be the reason? should batch4 wait until batch2,3 committed?

**Comment by qiwenzhou** [comment]
![img_v3_02b4_c183f4fe-4b21-40ee-934c-2c592fa7d2hu](https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/318927b4-795a-4337-b3fa-dcd9b28c290c)
I used block_reverter and restarted the server. what is next move to fix it according to the log?

**Comment by qiwenzhou** [comment]
<img width="886" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/f5b555c0-9a85-412e-87df-fff0f7110a15">
I ran：zk f cargo run --bin block_reverter --release -- rollback-db --l1-batch-number 3 --rollback-postgres --rollback-tree
and restart server

**Reply by bxpana** [reply]
Thanks for the additional info. Which set of instructions were you following to run your hyperchain?

**Reply by qiwenzhou** [reply]
thx for reply ! i ran on sepolia with erc20 and weth on and other instructions are by default.
what should i do under this circumstances? any suggestion？

**Reply by qiwenzhou** [reply]
maybe change aome data in db manually?

**Reply by bxpana** [reply]
Can you share the link of the documentation you used to set this up so we can try to replicate exactly as you did?

**Reply by qiwenzhou** [reply]
Sadlly there is no documentation for now. I try to write it down here for those i can remember.
1. hyperchain:
 1) zk && zk stack init:name,id,sepolia,L1 rpc, erc20 yes,weth no, use my own account to deploy、govern and operator.
 2) zk server --components=api,eth,tree,state_keeper,housekeeper,proof_data_handler,commitment_generator
 3) zk stack prover-setup && cd prover && ./setup gpu. zk f cargo run --release --bin zksync_prover_fri_gateway;API_PROMETHEUS_LISTENER_PORT=3116 zk f cargo run --release --bin zksync_witness_generator -- --all_rounds;FRI_WITNESS_VECTOR_GENERATOR_PROMETHEUS_LISTENER_PORT=3420 zk f cargo run --release --bin zksync_witness_vector_generator;zk f cargo run --features "gpu" --release --bin zksync_prover_fri;zk f cargo run --release --bin zksync_proof_fri_compressor
2. transactions
 1) I used default operator_blobs_eth_addr to bridge 0.01 eth to L2. This may cause  "nonce to low" problem, I guess?
 2) more deposits after that. 
 3) several hours later, server crushed. Direct reason is there is failure in L1 
![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/dbaaa4bd-9b6a-465a-b50d-8c5f40f7e446)
3. attempts
 1) zk f cargo run --bin block_reverter -- clear-failed-transactions
 2) zk f cargo run --bin block_reverter --rollback-db --l1_batch_number 2 --rollback_postgres --rollback_tree
 3) now when i try to restart server
![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/f7ce0981-237b-4ddf-b9a9-4bde420bf28b)

**Reply by qiwenzhou** [reply]
I changed the default_state_keeper_db_path and ran: k f cargo run --bin block_reverter --release -- rollback-db --l1-batch-number 2 --rollback-postgres --rollback-tree --rollback-sk-cache. Restart the server.
now the  error output is :
<img width="987" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/39e03224-c030-4dad-a45a-f0c8cfa33e3c">
How to update?

**Reply by qiwenzhou** [reply]
@bxpana would you please tell me that which db data should i change to fix this error?

**Reply by bxpana** [reply]
Are you still having issues with this?

**Reply by qiwenzhou** [reply]
Thanks for asking. This issue is not bothering me for now.
